### ISSUE TITLE
JSONProperty not working correctly in immutable models

### ISSUE THREAD

**Issue by paulhickman-a365** [domaindrivendev/Swashbuckle.AspNetCore#2057]
When using NewtonsSoft serialization, ASP.Net Core supports model binding using immutable models in controller methods. To work, this requires that the model has a constructor which has an argument for each JSON property with the same name as the property name (allowing a camel case vs. title case difference). This is also compatible with using ````[JsonProperty]```` to rename a property - the constructor has an argument with the real property name but it binds to the name specified in the ````[JsonProperty]````.

In the following controller, the two methods ````Mutable```` and ````Immutable```` both have the same JSON signature:

````c#
    [ApiController]
    [Route("Test")]
    public class TestController : ControllerBase
    {
        public class MutableModel
        {
            public int Prop1 { get; set; }

            [JsonProperty("prop2")]
            public int PropTwo { get; set; }
        }

        [HttpPost]
        [Route("Mutable")]
        public IActionResult Mutable([FromBody] MutableModel request)
        {
            return request.Prop1 == 1 && request.PropTwo == 1 ? Ok() : StatusCode(400);
        }

        public class ImmutableModel
        {
            public ImmutableModel(int prop1, int propTwo)
            {
                Prop1 = prop1;
                PropTwo = propTwo;
            }

            public int Prop1 { get; }

            [JsonProperty("prop2")]
            public int PropTwo { get; set; }
        }

        [HttpPost]
        [Route("Immutable")]
        public IActionResult Immutable([FromBody] ImmutableModel request)
        {
            return request.Prop1 == 1 && request.PropTwo == 1 ? Ok() : StatusCode(400);
        }
    }
````
With NewtsonSoft serialization, both of these methods will accept the JSON POST body of:

````json
{
   "prop1": 1,
   "prop2": 1
}
````
However, the swagger.json generated is:
````json
{
  "openapi": "3.0.1",
  "info": {
    "title": "My API",
    "version": "v1"
  },
  "paths": {
    "/Test/Mutable": {
      "post": {
        "tags": [
          "Test"
        ],
        "requestBody": {
          "content": {
            "application/json-patch+json": {
              "schema": {
                "$ref": "#/components/schemas/MutableModel"
              }
            },
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/MutableModel"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/MutableModel"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/MutableModel"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "Success"
          }
        }
      }
    },
    "/Test/Immutable": {
      "post": {
        "tags": [
          "Test"
        ],
        "requestBody": {
          "content": {
            "application/json-patch+json": {
              "schema": {
                "$ref": "#/components/schemas/ImmutableModel"
              }
            },
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/ImmutableModel"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/ImmutableModel"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/ImmutableModel"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "Success"
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "ImmutableModel": {
        "type": "object",
        "properties": {
          "prop1": {
            "type": "integer",
            "format": "int32"
          },
          "prop2": {
            "type": "integer",
            "format": "int32",
            "readOnly": true
          }
        },
        "additionalProperties": false
      },
      "MutableModel": {
        "type": "object",
        "properties": {
          "prop1": {
            "type": "integer",
            "format": "int32"
          },
          "prop2": {
            "type": "integer",
            "format": "int32"
          }
        },
        "additionalProperties": false
      }
    }
  }
}
````
Swashbuckle isn't allowing for the ````[JsonProperty]```` binding logic and is incorrectly marking ````prop2```` as read-only.

Example project: [SwashTest.zip](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/files/6132215/SwashTest.zip)

**Comment by domaindrivendev** [comment]
Have you followed configured Swashbuckle to honor Newtonsoft behavior instead of STJ (the default) as described in the readme?
https://github.com/domaindrivendev/Swashbuckle.AspNetCore#systemtextjson-stj-vs-newtonsoft

Also, your example link is broken

**Comment by paulhickman-a365** [comment]
Yes, Swashbuckle is configured correctly - you can see that ````prop2```` rather than ```propTwo```` appears in the swagger JSON which requires that it is setup to use NewtonSoft, and understand ````[JsonProperty]````

The problem is that the Swashbuckle anlayzer for NewtonSoft doesn't allow for using ````JsonProperty```` to rename a property when matching that property name to the constructor argument name to determine if it is settable.

I'll try attaching the project again here: 
[SwashTest.zip](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/files/6140175/SwashTest.zip)

**Comment by domaindrivendev** [comment]
OK dicovered the issue and fixed with a6eda1a40b5261efa6496ad10f685199e93a3778. Should now be available as of `6.6.1`. Can you pull down the latest version and see if it resolves your issue

**Comment by paulhickman-a365** [comment]
This has resolved the issue - thanks for the prompt update.
